mwdb/web/src/commons/ui/ObjectTab.tsx

```diff
@@ -17,24 +19,59 @@ type Props = {
     icon?: IconProp;
     tab: string;
     label?: string | JSX.Element;
+    dropdownActions?: boolean;
 };
 
 export function ObjectTab(props: Props) {
     const context: TabContextValues = useTabContext();
 
     useEffect(() => {
         if (context.tab !== props.tab) return;
+
         context.setComponent(props.component);
-        context.setActions(
-            props.actions?.map((action, index) => {
-                return {
-                    ...action,
-                    key: index,
-                };
-            }) || []
-        );
+
+        const previewSwitchAction =
+            props.actions?.find(
+                (action) => action.type === PreviewSwitchAction
+            ) || null;
+
+        const otherActions =
+            props.actions?.filter(
+                (action) => action.type !== PreviewSwitchAction
+            ) || [];
+
+        const actionsArray: JSX.Element[] = [];
+        if (previewSwitchAction) {
+            actionsArray.push(
+                React.cloneElement(previewSwitchAction, {
+                    key: "preview-switch",
+                })
+            );
+        }
+        if (otherActions.length > 0 && props.dropdownActions !== false) {
```

Comment:
```suggestion
        if (otherActions.length > 0 && props.dropdownActions) {
```

Reply:
The original condition was preserved in order to handle cases where dropdownAction was not defined as true.